components/new_post/post_component.tsx

```diff
@@ -0,0 +1,498 @@
+import PostBodyAdditionalContent from 'components/post_view/post_body_additional_content';
+import PostMessageContainer from 'components/post_view/post_message_view';
+import * as Utils from 'utils/utils';
+import {getHistory} from 'utils/browser_history';
+
+import PostUserProfile from './user_profile';
+import PostOptions from './post_options';
+
+export type Props = {
+    post: Post;
+    teamId: string;
+    currentUserId: string;
+    compactDisplay?: boolean;
+    colorizeUsernames?: boolean;
+    isFlagged: boolean;
+    isBusy?: boolean;
+    previewCollapsed?: string;
+    previewEnabled?: boolean;
+    isEmbedVisible?: boolean;
+    enableEmojiPicker?: boolean;
+    enablePostUsernameOverride?: boolean;
+    isReadOnly?: boolean;
+    pluginPostTypes?: {[postType: string]: PostPluginComponent};
+    channelIsArchived?: boolean;
+    isConsecutivePost?: boolean;
+    isLastPost?: boolean;
+    recentEmojis: Emoji[];
+    handleCardClick?: (post: Post) => void;
+    togglePostMenu?: (opened: boolean) => void;
+    channelName?: string;
+    displayName: string;
+    teamDisplayName?: string;
+    teamName?: string;
+    channelType?: string;
+    a11yIndex?: number;
+    isBot: boolean;
+    hasReplies?: boolean;
+    isFirstReply?: boolean;
+    previousPostIsComment?: boolean;
+    matches?: string[];
+    term?: string;
+    isMentionSearch?: boolean;
+    location: keyof typeof Locations;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
+        setActionsMenuInitialisationState: (viewed: Record<string, boolean>) => void;
+        selectPost: (post: Post) => void;
+        removePost: (post: Post) => void;
+        closeRightHandSide: () => void;
+        selectPostCard: (post: Post) => void;
+        setRhsExpanded: (rhsExpanded: boolean) => void;
+    };
+    timestampProps?: Partial<TimestampProps>;
+    shouldHighlight?: boolean;
+    isPostBeingEdited?: boolean;
+    isCollapsedThreadsEnabled?: boolean;
+    isMobileView: boolean;
+    canReply?: boolean;
+    replyCount?: number;
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const isSearchResultItem = (props.matches && props.matches.length > 0) || props.isMentionSearch || (props.term && props.term.length > 0);
+    const postRef = useRef<HTMLDivElement>(null);
+    const postHeaderRef = useRef<HTMLDivElement>(null);
+    const [hover, setHover] = useState(false);
+    const [a11yActive, setA11y] = useState(false);
+    const [dropdownOpened, setDropdownOpened] = useState(false);
+    const [fileDropdownOpened, setFileDropdownOpened] = useState(false);
+    const [fadeOutHighlight, setFadeOutHighlight] = useState(false);
+    const [alt, setAlt] = useState(false);
+
+    if (props.shouldHighlight) {
+        setTimeout(() => {
+            setFadeOutHighlight(true);
+        }, Constants.PERMALINK_FADEOUT);
+    }
```

Comment:
we are not clearing the previous set time out here. leads to leaks

Reply:
nice catch!

Reply:
nice catch!